modules/basegl/src/processors/imageprocessing/columnrowlayout.cpp

```diff
@@ -0,0 +1,325 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/basegl/processors/imageprocessing/columnrowlayout.h>
+
+#include <inviwo/core/util/assertion.h>
+#include <inviwo/core/network/networklock.h>
+#include <modules/opengl/texture/textureutils.h>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo ColumnLayout::processorInfo_{
+    "org.inviwo.ColumnLayout",  // Class identifier
+    "Column Layout",            // Display name
+    "Image Operation",          // Category
+    CodeState::Stable,          // Code state
+    Tags::GL | Tag("Layout"),   // Tags
+};
+const ProcessorInfo ColumnLayout::getProcessorInfo() const { return processorInfo_; }
+
+const ProcessorInfo RowLayout::processorInfo_{
+    "org.inviwo.RowLayout",    // Class identifier
+    "Row Layout",              // Display name
+    "Image Operation",         // Category
+    CodeState::Stable,         // Code state
+    Tags::GL | Tag("Layout"),  // Tags
+};
+const ProcessorInfo RowLayout::getProcessorInfo() const { return processorInfo_; }
+
+Layout::Layout(splitter::Direction direction)
+    : Processor()
+    , inport_("inport")
+    , outport_("outport")
+    , splitterSettings_("splitterSettings", "Style", true, splitter::Style::Line,
+                        vec4(0.27f, 0.3f, 0.31f, 1.0f), vec4(0.1f, 0.1f, 0.12f, 1.0f))
+    , minWidth_("minWidth", "Minimum Width (px)", 10, 0, 4096, 1, InvalidationLevel::InvalidOutput,
+                PropertySemantics::SpinBox)
+    , splitters_("splitters", "Split Positions")
+    , renderer_(this)
+    , direction_(direction)
+    , currentDim_(0, 0)
+    , shader_("img_texturequad.vert", "img_copy.frag")
+    , deserialized_(false) {
+
+    addPort(inport_);
+    addPort(outport_);
+
+    splitterSettings_.setCollapsed(true);
+    splitterSettings_.width_.set(4.0f);
+    splitterSettings_.triSize_.set(0.0f);
+    splitterSettings_.setCurrentStateAsDefault();
+
+    addProperties(splitterSettings_, minWidth_, splitters_);
+
+    inport_.setIsReadyUpdater([this]() {
+        // Ports with dimensions (1,1) or less are considered to be inactive
```

Comment:
you say less here but use equal 1 below?

Reply:
good catch. Should be addressed now.